pkg/snowflake/operator.go

```diff
@@ -94,7 +95,19 @@ func (o BasicOperator) RunTask(ctx context.Context, p *pipeline.Pipeline, t *pip
 			return err
 		}
 	}
-	return conn.RunQueryWithoutResult(ctx, q)
+	// Run the main query
+	err = conn.RunQueryWithoutResult(ctx, q)
+	if err != nil {
+		return err
+	}
+
+	if p.MetadataPush.HasAnyEnabled() {
+		err = conn.PushColumnDescriptions(ctx, t)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
```

Comment:
we shouldn't do it here, instead this should be a new operator that is executed outside of the main asset, otherwise we'll be rerunning the query when the user wants to only push the metadata. take a look at the bigquery implementation for an inspiration.